pkg/controller/plan/kubevirt.go

```diff
@@ -857,6 +859,48 @@ func (r *KubeVirt) GetGuestConversionPod(vm *plan.VMStatus) (pod *core.Pod, err
 	return
 }
 
+func (r *KubeVirt) GetVirtV2VDiskConfig(vm *plan.VMStatus, pod *core.Pod, step *plan.Step) (err error) {
+
+	if pod.Status.PodIP == "" {
+		//we need the IP for fetching the fimware info.
+		return
+	}
+
+	url := fmt.Sprintf("http://%s:8080/firmware", pod.Status.PodIP)
+	resp, err := http.Get(url)
+
+	if err != nil {
+		if strings.Contains(err.Error(), "connection refused") {
+			return nil
+		}
+		return
+	} else {
+		defer resp.Body.Close()
+		firmaware, errFirrmware := io.ReadAll(resp.Body)
+		fmt.Println("we are here firmware ", firmaware)
+		if errFirrmware != nil {
+			return errFirrmware
+		}
+
+		currentAnnotations := r.Migration.Annotations
+		if currentAnnotations == nil {
+			currentAnnotations = make(map[string]string)
+		}
+		var vmId string
+		labels := pod.GetLabels()
+		if val, ok := labels["vmID"]; ok {
+			vmId = val
+		}
+		currentAnnotations[vmId] = string(firmaware)
+		r.Migration.SetAnnotations(currentAnnotations)
+		err = r.Destination.Client.Update(context.TODO(), r.Migration.DeepCopy())
```

Comment:
`patch` seems to better fit here
but why placing it in the annotations in the first place (and not in the status)?

Reply:
//TODO

Reply:
@bkhizgiy  did you check whether this call to `Update` triggers a reconcile of the migration?

Reply:
nope, I just saw it was correctly added, but now that I've moved this change to the vmStatus inside the migration I cant get it to update correctly, its not being saved.

Reply:
I'm not sure why but the status isnt updated in the CR, tried from both the migration and plan objects. for now I returned it to annotation, and once I"ll finish with the ova validation i"ll fix it.

I was thinking of adding a new map under the VM status, and there we can save all the needed config for this vm, so we wont need to have map of the vm id and the resource.

Reply:
> I was thinking of adding a new map under the VM status, and there we can save all the needed config for this vm, so we wont need to have map of the vm id and the resource.

yeah, that would be a better approach
we don't even need a new map, we can have something like:
``` yaml
status:
  ...
  vms:
  - id: ...
    name: ...
    ovf: <ovf>
```